interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/Annotations.kt

```diff
@@ -21,31 +21,28 @@ import com.google.devtools.ksp.symbol.KSAnnotation
 import com.google.devtools.ksp.symbol.KSClassDeclaration
 import com.google.devtools.ksp.symbol.KSName
 import com.google.devtools.ksp.symbol.KSType
-import com.google.devtools.ksp.symbol.KSTypeAlias
 import com.squareup.kotlinpoet.AnnotationSpec
 import com.squareup.kotlinpoet.AnnotationSpec.UseSiteTarget
 import com.squareup.kotlinpoet.ClassName
 import com.squareup.kotlinpoet.CodeBlock
-import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.ParameterizedTypeName
 
 /**
  * Returns an [AnnotationSpec] representation of this [KSAnnotation] instance.
  * @param omitDefaultValues omit defining default values when `true`
  */
 @JvmOverloads
 public fun KSAnnotation.toAnnotationSpec(omitDefaultValues: Boolean = false): AnnotationSpec {
-  val builder = annotationType.resolve().unwrapTypeAlias().toClassName()
-    .let { className ->
-      val typeArgs = annotationType.element?.typeArguments.orEmpty()
-        .map { it.toTypeName() }
-      if (typeArgs.isEmpty()) {
-        AnnotationSpec.builder(className)
-      } else {
-        AnnotationSpec.builder(className.parameterizedBy(typeArgs))
-      }
-    }
+  val typeName = annotationType.resolve().toTypeName()
+
+  val builder = if (typeName is ClassName) {
+    AnnotationSpec.builder(typeName)
+  } else {
+    AnnotationSpec.builder(typeName as ParameterizedTypeName)
+  }
```

Comment:
If `typeName` is neither `ClassName` nor `ParameterizedTypeName` - we crash?

Reply:
Yes but it should be impossible for that to happen here